common/persistence/pinot/pinot_visibility_store.go

```diff
@@ -721,19 +721,17 @@ func (s *PinotQuerySearchField) resetSearchField() {
 func (s *PinotQuerySearchField) addEqual(obj string, val interface{}) {
 	s.checkFirstSearchField()
 	if _, ok := val.(string); ok {
-		val = fmt.Sprintf("'%s'", val)
+		s.string += fmt.Sprintf("%s = '%s'\n", obj, val)
```

Comment:
The whole thing with += of strings is extremely error-prone and inefficient.

I suggest revisiting this and building a set of filters to groups with "()" around them and then joining them together in one go.

But that would be a separate problem/PR.

cc @neil-xie